### DISCUSSION TITLE
Issue with download_directory and tempdir

### DISCUSSION THREAD

**Discussion by jkaucic** [ropensci/osmextract#274] [Q&A] [answered]
I have set my download_directory to an external ssd and the vectortranslate operation is creating a gpkg on this ssd drive just fine. However, as written here: https://gdal.org/drivers/vector/osm.html#vector-osm
"The driver will use an internal SQLite database to resolve geometries. If that database remains under 100 MB it will reside in RAM. If it grows above, it will be written in a temporary file on disk. By default, this file will be written in the current directory, unless you define the CPL_TMPDIR configuration option."
This leads to the writing of huge temporary files in my C:\Users\xxx\AppData\Local\Temp folder, which - opposed to the statement of default writing into current directory - is written in the temp folder and not on the SSD.
It seems that the configuration of the download directory in R osmextract / r.environ doesn't alter the GDAL current directory.
Do you know how I can set the configuration option of GDAL CPL_TMPDIR=OFF so it writes temporary files also on my SSD in  the download directory there? 
Thank you

**Comment by agila5** [comment, marked as answer]
Hi! I think you should run `Sys.setenv(...)` at the beginning of your code. For example, 

``` r
library(sf)
#> Linking to GEOS 3.11.1, GDAL 3.6.2, PROJ 9.1.1; sf_use_s2() is TRUE
osm_path <- system.file("osm/overpass.osm", package="sf")
st_layers(osm_path, do_count = TRUE)
#> Warning in CPL_get_layers(dsn, options, do_count): GDAL Error 1: Non increasing
#> node id. Use OSM_USE_CUSTOM_INDEXING=NO

#> ...
#>...

#> Warning in CPL_get_layers(dsn, options, do_count): GDAL Error 1: Non increasing
#> node id. Use OSM_USE_CUSTOM_INDEXING=NO

#> Driver: OSM 
#> Available layers:
#>         layer_name       geometry_type features fields crs_name
#> 1           points               Point        0     10   WGS 84
#> 2            lines         Line String        0     10   WGS 84
#> 3 multilinestrings   Multi Line String        0      4   WGS 84
#> 4    multipolygons       Multi Polygon        3     25   WGS 84
#> 5  other_relations Geometry Collection        0      4   WGS 84
Sys.setenv(OSM_USE_CUSTOM_INDEXING="NO")
st_layers(osm_path, do_count = TRUE)
#> Driver: OSM 
#> Available layers:
#>         layer_name       geometry_type features fields crs_name
#> 1           points               Point        1     10   WGS 84
#> 2            lines         Line String        0     10   WGS 84
#> 3 multilinestrings   Multi Line String        0      4   WGS 84
#> 4    multipolygons       Multi Polygon       13     25   WGS 84
#> 5  other_relations Geometry Collection        0      4   WGS 84
```

<sup>Created on 2023-02-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Let me know if it works also in your case!

**Reply by jkaucic** [reply]
Thank you! Putting this code before my oe_get command worked - i just defined my oe_download_directory() also as CPL_TMPDIR:

`Sys.setenv(CPL_TMPDIR="D:\\23-02-16_OSM_Geofabrik_Europe")`

Now, the intermediate osm_tmp_nodes_xxx file and the final .gpkg output are all written to the directory I have set.

**Comment by agila5** [comment]
The development version of `sf` implemented a simpler approach to solve this problem: https://github.com/r-spatial/sf/issues/2003 I will document it here as soon as the new version is released on CRAN.
